src/transformers/models/llama/modeling_llama.py

```diff
@@ -384,11 +409,193 @@ def forward(
         return attn_output, attn_weights, past_key_value
 
 
+class LlamaFlashAttention2(nn.Module):
```

Comment:
same controversial comment here, for readability I'm in favor of inheriting from the `LlamaAttention` and overriding the forward pass. It should be contained to a single level of inheritance however.

Reply:
Agreed, I think that way it will make things much more readable and maintainable